### PR TITLE
Fix JSON rendering on json recursion exception

### PR DESCRIPTION
**Pull request by thenotsoft**

### CODE REVIEW

File: src/Renderer/JsonRenderer.php [hunk 1]
```diff
@@ -19,20 +19,30 @@ final class JsonRenderer implements ThrowableRendererInterface
...
+            json_encode(
+                [
+                    'type' => get_class($t),
+                    'message' => $t->getMessage(),
+                    'code' => $t->getCode(),
+                    'file' => $t->getFile(),
+                    'line' => $t->getLine(),
+                    'trace' => $t->getTrace(),
+                ],
+                JSON_THROW_ON_ERROR | JSON_PRETTY_PRINT | JSON_UNESCAPED_SLASHES | JSON_INVALID_UTF8_SUBSTITUTE | JSON_PARTIAL_OUTPUT_ON_ERROR
```
Comment (vjik) [inline src/Renderer/JsonRenderer.php:44]:
```suggestion
                JSON_PRETTY_PRINT | JSON_UNESCAPED_SLASHES | JSON_INVALID_UTF8_SUBSTITUTE | JSON_PARTIAL_OUTPUT_ON_ERROR
```

[PHP docs](https://www.php.net/manual/en/json.constants.php):
> JSON_PARTIAL_OUTPUT_ON_ERROR takes precedence over JSON_THROW_ON_ERROR

File: tests/Renderer/JsonRendererTest.php [hunk 2]
```diff
@@ -53,6 +53,40 @@ public function testRenderVerboseWithNotUtf8String(): void
...
+                    'message' => $throwable->getMessage(),
+                    'code' => $throwable->getCode(),
+                    'file' => $throwable->getFile(),
+                    'line' => $throwable->getLine(),
+                    'trace' => $throwable->getTrace(),
+                ],
+                JSON_PRETTY_PRINT | JSON_UNESCAPED_SLASHES | JSON_INVALID_UTF8_SUBSTITUTE | JSON_PARTIAL_OUTPUT_ON_ERROR
+            );
+
+            $this->assertSame($content, (string)$data);
```
Comment (vjik) [inline tests/Renderer/JsonRendererTest.php:89]: Assertion better to move out of from `try ... catch`.

**Review by vjik — APPROVED** [review]

**Comment by vjik** [comment]
Need line in changelog

**Comment by samdark** [comment]
👍

**Pull request merged**